pkg/app/piped/imagewatcher/watcher.go

```diff
@@ -196,21 +203,31 @@ func (w *watcher) updateOutdatedImage(ctx context.Context, target *config.ImageW
 	}
 
 	// Update the outdated image.
+	//
+	// Clone repo into another directory to avoid pull failure in the future.
+	tmpDir, err := ioutil.TempDir("", "image-watcher")
+	if err != nil {
+		return fmt.Errorf("failed to create a new temporary directory: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	repo, err := w.gitClient.Clone(ctx, repoCfg.RepoID, repoCfg.Remote, repoCfg.Branch, tmpDir)
```

Comment:
`repo.Copy` may be better. Because cloning is retrieving a new git tree. The commit-tree could be added from the last pull.

Reply:
Yup, definitely right.